### PR TITLE
FFT bugfixes

### PR DESCRIPTION
**Pull request by andyp1per**
This fixes four bugs related to the FFT support:

- Check for 1 rather than 0 in Quinn's estimator to avoid a possible divide by zero
- Make sure that user specified IMU noise is present if RPM noise is not being used
- The autotest for harmonic matching is a little flaky, I have relaxed the constraints pending looking into the cause
- Changed default window size on H7 boards to 64. This is because they can easily cope and generally have BMI088 which has a backend rate of 2kHz leading to poor frequency resolution at short lengths

### CODE REVIEW

**Comment by andyp1per** [comment]
@rmackay9 I checked that with your test we now get sensor noise:

image

I wasn't able to exactly reproduce your 3.6 scenario so not sure if there is something else going on as well.

File: libraries/AP_HAL/DSP.cpp [hunk 1]
```diff
@@ -202,7 +202,7 @@ float DSP::calculate_quinns_second_estimator(const FFTWindowState* fft, const fl
     const float am = (complex_fft[k_m1] * complex_fft[k] + complex_fft[k_m1 + 1] * complex_fft[k + 1]) / divider;
 
     // sanity check
-    if (is_zero(ap) || is_zero(am)) {
+    if (is_equal(ap, 1.0f) || is_equal(am, 1.0f)) {
```
Comment (tridge) [inline libraries/AP_HAL/DSP.cpp]: I think it should check if (1.0 - ap) is below some minimum threshold, same for am

Comment (andyp1per) [reply libraries/AP_HAL/DSP.cpp]: Ok fixed. I'm not sure what an appropriate lower bound is, but I have some real values in a spreadsheet where am and ap are between 0.1 and 0.9 so 0.01 seems pretty safe.

**Comment by andyp1per** [comment]
There is a slight problem with the SITL noise fix, it adds noise when a single freq is set which is not what used to happen. Will push a fix shortly.

**Comment by andyp1per** [comment]
Ok fixed

**Comment by andyp1per** [comment]
@tridge I think the fix I just pushed will fix your self-test issue. You should set FFT_LENGTH to 64 which is now the default for the Durandal.

**Comment by andyp1per** [comment]
But was I right @tridge?! It's killing me not knowing 😉

**Pull request merged**